crates/wasmi/src/memory/buffer.rs

```diff
@@ -20,33 +28,151 @@ impl ByteBuffer {
     /// - If the initial length is 0.
     /// - If the initial length exceeds the maximum supported limit.
     pub fn new(initial_len: usize) -> Self {
+        let mut vec = ManuallyDrop::new(vec![0x00_u8; initial_len]);
+        let (ptr, len, capacity) = (vec.as_mut_ptr(), vec.len(), vec.capacity());
         Self {
-            bytes: vec![0x00_u8; initial_len],
+            ptr,
+            len,
+            capacity,
+            is_static: false,
+        }
+    }
+
+    /// Creates a new byte buffer with the given initial length.
+    ///
+    /// # Errors
+    ///
+    /// - If the initial length is 0.
+    /// - If the initial length exceeds the maximum supported limit.
+    pub fn new_static(buf: &'static mut [u8], initial_len: usize) -> Self {
+        assert!(initial_len <= buf.len());
+        buf.fill(0);
```

Comment:
Only fill `buf[0..initial_len]`

Reply:
Is there any benefit to increase complexity?

Reply:
Yes, for the sake of fuel metering we charge linearly for `memory.grow` anyways, so it is wise to actually distribute the work to zero the memory instead of doing it up front which might be not desirable for large static buffers in all cases.